storage_v2/factoryadapter/reader.go

```diff
@@ -71,30 +78,39 @@ func (tr *TraceReader) GetOperations(ctx context.Context, query tracestore.Opera
 func (tr *TraceReader) FindTraces(
 	ctx context.Context,
 	query tracestore.TraceQueryParameters,
-) ([]ptrace.Traces, error) {
-	t, err := tr.spanReader.FindTraces(ctx, query.ToSpanStoreQueryParameters())
-	if err != nil || t == nil {
-		return nil, err
+) iter.Seq2[[]ptrace.Traces, error] {
+	return func(yield func([]ptrace.Traces, error) bool) {
+		t, err := tr.spanReader.FindTraces(ctx, query.ToSpanStoreQueryParameters())
+		if err != nil || t == nil {
+			yield(nil, err)
+			return
+		}
+		for _, trace := range t {
+			batch := &model.Batch{Spans: trace.GetSpans()}
+			otelTrace, _ := model2otel.ProtoToTraces([]*model.Batch{batch})
+			if !yield([]ptrace.Traces{otelTrace}, nil) {
+				return
+			}
```

Comment:
is it possible to test this block? any reason we can't just always yield?

Reply:
Should be possible to test. And we do always yield, but we have to respect the contract of Seq2 - if yield lambda returns false the iterator must stop producing values.